chain/chain/src/validate.rs

```diff
@@ -161,25 +161,25 @@ fn validate_congestion_info(
 }
 
 fn validate_bandwidth_requests(
-    extra_bandwidth_requests: Option<&BandwidthRequests>,
+    extra_bandwidth_requests: &BandwidthRequests,
     header_bandwidth_requests: Option<&BandwidthRequests>,
 ) -> Result<(), Error> {
-    if extra_bandwidth_requests != header_bandwidth_requests {
-        fn requests_len(requests_opt: Option<&BandwidthRequests>) -> usize {
-            match requests_opt {
-                Some(BandwidthRequests::V1(requests_v1)) => requests_v1.requests.len(),
-                None => 0,
-            }
+    if let Some(header_bandwidth_requests) = header_bandwidth_requests {
+        if extra_bandwidth_requests == header_bandwidth_requests {
+            return Ok(());
+        }
+        fn requests_len(BandwidthRequests::V1(requests_v1): &BandwidthRequests) -> usize {
+            requests_v1.requests.len()
```

Comment:
this can also be a bit more explicit if we keep the deconstruction in a match or a bit more short if we make it a lambda and infer the return type.